go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -231,20 +232,16 @@ func (oc *Controller) syncEgressFirewall(egressFirwalls []interface{}) {
 
 func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
 	klog.Infof("Adding egressFirewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
-	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
-	if err != nil {
-		return fmt.Errorf("failed to wait for namespace %s event (%v)",
-			egressFirewall.Namespace, err)
-	}
-	defer nsInfo.Unlock()
-
-	if nsInfo.egressFirewall != nil {
+	oc.egressFirewallsMutex.Lock()
+	defer oc.egressFirewallsMutex.Unlock()
+	if _, exists := oc.egressFirewalls[egressFirewall.Namespace]; exists {
```

Comment:
in the delete you check if egressFirewalls value could be nil. Is it worth checking here as well? If a egressFirewalls[egressFirewall.Namespace] exists, but is nil, should we continue and not report error? Or is that an error?

Reply:
Do I? it should never be nil?

Reply:
checked